server/src/main/java/io/druid/server/initialization/jetty/JettyServerModule.java

```diff
@@ -161,6 +162,10 @@ private static Server makeJettyServer(@Self DruidNode node, ServerConfig config)
 
     final Server server = new Server(threadPool);
 
+    // Without this bean set, the default ScheduledExecutorScheduler runs as non-daemon, causing lifecycle hooks to fail
+    // to fire on main exit. Related bug: https://github.com/druid-io/druid/pull/1627
+    server.addBean(new ScheduledExecutorScheduler("JettyScheduler", true), true);
```

Comment:
looks like I'm missing something, but dint we decide to call lifecycle.stop() explicitly and via shutdown hook in cases where Main is expected to exit voluntarily (which is in case of CliPeon only for now) ? In other CliXXX classes, shutdown hook is always registered and they can only exit on a signal which will call the shutdown hook?